Sources/Datadog/Logging/RemoteLogger.swift

```diff
@@ -44,6 +36,12 @@ internal final class RemoteLogger: LoggerProtocol {
     /// Integration with Tracing. It is used to correlate Logs with Spans by injecting `Span` context to `LogEvent`.
     /// Can be `false` if the integration is disabled for this logger.
     internal let activeSpanIntegration: Bool
+    /// Logger-specific attributes.
+    @_pthread_rwlock
+    private var attributes: [String: Encodable] = [:]
```

Comment:
nit/ Wouldn't it be cleaner if we make up more descriptive name for `@_pthread_rwlock`? Now it reflects the underlying implementation, but for someone who's an occasional user of `pthread` APIs (like I am 😄), this is always confusing. Each time I had to visit docs to ensure myself that this is _"concurrent reads, exclusive writes"_ lock.

Maybe this?
```
@AtomicConcurrentReads
```

WDYT @maxep @maciejburda ?

Reply:
This type of lock is usually referred as [read-write-lock](https://pubs.opengroup.org/onlinepubs/009604599/functions/pthread_rwlock_rdlock.html), we could name this wrapper `ReadWriteLock` instead (that's the name used in java)?

I agree `@_pthread_rwlock` is not clear, but it looks cool 😄

Reply:
👍 for `ReadWriteLock` then 🙂